src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -215,15 +222,24 @@ class F extends CopyToTextureUtils {
     gl.clearColor(colorValue, colorValue, colorValue, alphaValue);
     gl.clear(gl.COLOR_BUFFER_BIT);
 
-    return { canvas, canvasContext: gl };
+    return {
+      canvas,
+      expectedSourceData: this.getExpectedReadbackForWebGLCanvas(gl, width, height),
+    };
   }
 
-  getInitGPUCanvasData(width: number, height: number, premultiplied: boolean): Uint8ClampedArray {
+  private getDataToInitSourceWebGPUCanvas(
+    width: number,
+    height: number,
+    alphaMode: GPUCanvasAlphaMode
+  ): Uint8ClampedArray {
     const rectWidth = Math.floor(width / 2);
     const rectHeight = Math.floor(height / 2);
 
     const alphaValue = 153;
-    const colorValue = premultiplied ? alphaValue : 255;
+    // colorValue can just always = alphaValue now, because readback will clear it to
+    // 255 if opaque.
```

Comment:
I think I misunderstood a little bit when I was writing this comment, but anyway it's not entirely relevant anymore. Here's a more clarifying comment, if you think it makes sense:

```suggestion
    // Always output [153, 153, 153, 153]. When the alphaMode is...
    //   - premultiplied: the readback is CSS `rgba(255, 255, 255, 60%)`.
    //   - opaque: the readback is CSS `rgba(153, 153, 153, 100%)`.
    // getExpectedReadbackForWebGPUCanvas matches this.
```